libchisel/src/repack.rs

```diff
@@ -0,0 +1,124 @@
+
+impl Repack {
+    pub fn new() -> Self {
+        Repack {}
+    }
+}
+
+impl ModuleTranslator for Repack {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Err(ModuleError::NotSupported)
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let module = module.clone();
+        let module = builder::from_module(module).build();
+        Ok(Some(module))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use parity_wasm::builder;
+
+    // Smoke tests
+    #[test]
+    fn builder_all_exports_good_ewasm() {
+        let mut module = builder::module()
+            .function()
+            .signature()
+            .build()
+            .body()
+            .build()
+            .build()
+            .export()
+            .field("main")
+            .internal()
+            .func(0)
+            .build()
+            .export()
+            .field("memory")
+            .internal()
+            .memory(0)
+            .build()
+            .build();
+
+        let trimmer = TrimExports::with_preset("ewasm").unwrap();
+        let did_change = trimmer.translate_inplace(&mut module).unwrap();
+        assert_eq!(false, did_change);
+    }
+
+    #[test]
+    fn builder_one_wrong_mem_export_ewasm() {
+        let mut module = builder::module()
+            .function()
+            .signature()
+            .build()
+            .body()
+            .build()
+            .build()
+            .export()
+            .field("main")
+            .internal()
+            .func(0)
+            .build()
+            .export()
+            .field("memory")
+            .internal()
+            .memory(0)
+            .build()
+            .export()
+            .field("foo")
+            .internal()
+            .memory(0)
+            .build()
+            .build();
+
+        let trimmer = TrimExports::with_preset("ewasm").unwrap();
+        let did_change = trimmer.translate_inplace(&mut module).unwrap();
+        assert_eq!(true, did_change);
+    }
+
+    #[test]
+    fn builder_no_export_ewasm() {
+        let mut module = builder::module()
+            .function()
+            .signature()
+            .build()
+            .body()
+            .build()
+            .build()
+            .build();
+
+        let trimmer = TrimExports::with_preset("ewasm").unwrap();
+        let did_change = trimmer.translate_inplace(&mut module).unwrap();
+        assert_eq!(false, did_change);
+    }
+
+    #[test]
+    fn builder_all_exports_good_pwasm() {
+        let mut module = builder::module()
+            .function()
+            .signature()
+            .build()
+            .body()
+            .build()
+            .build()
+            .export()
+            .field("_call")
+            .internal()
+            .func(0)
+            .build()
+            .build();
+
+        let trimmer = TrimExports::with_preset("pwasm").unwrap();
```

Comment:
what's going on here with these tests?
From what I gather this should be repack.

Reply:
As I said above tests need to be finished. The source file was trimexports.